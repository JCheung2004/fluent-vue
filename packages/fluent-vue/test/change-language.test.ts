import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../src'

describe('language change', () => {
  let options: any
  let bundleEn: FluentBundle
  let bundleUk: FluentBundle

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(FluentVue)

    bundleEn = new FluentBundle('en-US')
    bundleUk = new FluentBundle('uk-UA')

    const fluent = new FluentVue({
      bundles: [bundleUk, bundleEn]
    })

    options = {
      fluent,
      localVue
    }
  })

  it('can work with multiple bundles', () => {
    // Arrange
    bundleEn.addResource(
      new FluentResource(ftl`
      link = link text
      `)
    )

    bundleUk.addResource(
      new FluentResource(ftl`
      link = текст посилання
      `)
    )

    const component = {
      template: `<a v-t:link href="/foo">Fallback text</a>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('falls back to previous bundle', () => {
    // Arrange
    bundleEn.addResource(
      new FluentResource(ftl`
      link = link text
      `)
    )

    const component = {
      template: `<a v-t:link href="/foo">Fallback text</a>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('updates when updating bundles array', () => {
    // Arrange
    const localVue = createLocalVue()
    localVue.use(FluentVue)

    bundleEn = new FluentBundle('en-US')
    bundleUk = new FluentBundle('uk-UA')

    const fluent = new FluentVue({
      bundles: [bundleUk, bundleEn]
    })

    bundleEn.addResource(
      new FluentResource(ftl`
      link = link text
      `)
    )

    bundleUk.addResource(
      new FluentResource(ftl`
      link = текст посилання
      `)
    )

    const component = {
      template: `<a v-t:link href="/foo">Fallback text</a>`
    }

    // Act
    const mounted = mount(component, {
      fluent,
      localVue
    })

    expect(mounted).toMatchSnapshot()

    fluent.bundles = [bundleEn, bundleUk]

    // Assert
    expect(mounted).toMatchSnapshot()
  })
})
