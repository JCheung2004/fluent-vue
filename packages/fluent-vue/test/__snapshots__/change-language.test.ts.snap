// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language change can work with multiple bundles 1`] = `<a href="/foo">текст посилання</a>`;

exports[`language change falls back to previous bundle 1`] = `<a href="/foo">link text</a>`;

exports[`language change updates when updating bundles array 1`] = `<a href="/foo">текст посилання</a>`;

exports[`language change updates when updating bundles array 2`] = `<a href="/foo">link text</a>`;
